lib/client/hostendpoint_e2e_test.go

```diff
@@ -0,0 +1,303 @@
+				Profiles:      []string{"profile3", "profile4"},
+			}),
+
+		// Test 2: Pass one partially populated HostEndpointSpec and another fully populated HostEndpointSpec and expect the series of operations to succeed.
+		Entry("One partially populated HostEndpointSpec and another fully populated HostEndpointSpec",
+			api.HostEndpointMetadata{
+				Name: "host1",
+				Node: "hostname1",
+				Labels: map[string]string{
+					"app":  "app-abc",
+					"prod": "no",
+				}},
+			api.HostEndpointMetadata{
+				Name: "host2",
+				Node: "hostname2",
+				Labels: map[string]string{
+					"app":  "app-xyz",
+					"prod": "yes",
+				}},
+			api.HostEndpointSpec{
+				InterfaceName: "eth0",
+			},
+			api.HostEndpointSpec{
+				InterfaceName: "eth1",
+				ExpectedIPs:   []cnet.IP{testutils.MustParseIP("192.168.0.0"), testutils.MustParseIP("192.168.1.1")},
+				Profiles:      []string{"profile3", "profile4"},
+			}),
+
+		// Test 3: Pass one fully populated HostEndpointSpec and another empty HostEndpointSpec and expect the series of operations to succeed.
+		Entry("One fully populated HostEndpointSpec and another empty HostEndpointSpec",
+			api.HostEndpointMetadata{
+				Name: "host1",
+				Node: "hostname1",
+				Labels: map[string]string{
+					"app":  "app-abc",
+					"prod": "no",
+				}},
+			api.HostEndpointMetadata{
+				Name: "host2",
+				Node: "hostname2",
+				Labels: map[string]string{
+					"app":  "app-xyz",
+					"prod": "yes",
+				}},
+			api.HostEndpointSpec{
+				InterfaceName: "eth0",
+				ExpectedIPs:   []cnet.IP{testutils.MustParseIP("10.0.0.0"), testutils.MustParseIP("20.0.0.0")},
+				Profiles:      []string{"profile1", "profile2"},
+			},
+			api.HostEndpointSpec{}),
+
+		// Test 4: Pass two fully populated HostEndpointSpecs with two HostEndpointMetadata (one IPv4 and another IPv6) and expect the series of operations to succeed.
+		Entry("Two fully populated HostEndpointSpecs with two HostEndpointMetadata (one IPv4 and another IPv6)",
+			api.HostEndpointMetadata{
+				Name: "host1",
+				Node: "hostname1",
+				Labels: map[string]string{
+					"app":  "app-abc",
+					"prod": "no",
+				}},
+			api.HostEndpointMetadata{
+				Name: "host2",
+				Node: "hostname2",
+				Labels: map[string]string{
+					"app":  "app-xyz",
+					"prod": "yes",
+				}},
+			api.HostEndpointSpec{
+				InterfaceName: "eth0",
+				ExpectedIPs:   []cnet.IP{testutils.MustParseIP("10.0.0.0"), testutils.MustParseIP("192.168.1.1")},
+				Profiles:      []string{"profile1", "profile2"},
+			},
+			api.HostEndpointSpec{
+				InterfaceName: "eth1",
+				ExpectedIPs:   []cnet.IP{testutils.MustParseIP("fe80::00"), testutils.MustParseIP("fe80::33")},
+				Profiles:      []string{"profile3", "profile4"},
+			}),
+	)
+
+})
+
+// hostEndpointCreate takes client and api.HostEndpoint and creates a HostEndpoint.
+func hostEndpointCreate(c *client.Client, p *api.HostEndpoint) (*api.HostEndpoint, error) {
```

Comment:
I thought we were getting rid of these methods?  That means across all the tests.